validator/keymanager/v2/derived/derived.go

```diff
@@ -226,6 +223,22 @@ func (dr *Keymanager) NextAccountNumber(ctx context.Context) uint64 {
 	return dr.seedCfg.NextAccount
 }
 
+// RecoverFromMnemonic given a mnemonic phrase, is able to regenerate a wallet seed.
+func (dr *Keymanager) RecoverFromMnemonic(ctx context.Context, mnemonic string) error {
```

Comment:
This function name could be misleading, it writes the seed file from disk and `RecoverFromMnemonic` might be too vague to let others know this. Maybe it could be rephrased?